gen/definitions/device.yaml

```diff
@@ -0,0 +1,97 @@
+---
+name: Device
+rest_endpoint: /api/fmc_config/v1/domain/{DOMAIN_UUID}/devices/devicerecords
+doc_category: Devices
+test_tags: [FTD_ADDR]
+attributes:
+  - model_name: name
+    type: String
+    mandatory: true
+    description: "User-specified name, must be unique. Example: 'Device 01 - 192.168.0.152'"
+    example: device1
+    minimum_test_value: '"device1alt"'
+  - model_name: hostName
+    type: String
+    description: Hostname or IP address of the device. Either the host_name or nat_id must be present.
+    example: 10.0.0.1
+    minimum_test_value: var.ftd_addr
+    test_value: var.ftd_addr
+  - model_name: natID
+    tf_name: nat_id
+    type: String
+    description: (used for device registration behind NAT) If the device to be registered and the Firepower Management Center are separated by network address translation (NAT), set a unique string identifier.
+    write_only: true
+    exclude_example: true
+    test_value: var.nat_id
+    minimum_test_value: var.nat_id
+  - model_name: license_caps
+    type: Set
+    element_type: String
+    mandatory: true
+    description: "Array of strings representing the license capabilities on the managed device. For registering FTD, the allowed values are: BASE (mandatory), THREAT, URLFilter, MALWARE, APEX, PLUS, VPNOnly. For Firepower ASA or NGIPSv devices, allowed values are: BASE, THREAT, PROTECT, CONTROL, URLFilter, MALWARE, VPN, SSL."
+    example: BASE
+    minimum_test_value: '["THREAT", "URLFilter", "BASE", "MALWARE"]'
+  - model_name: regKey
+    type: String
+    mandatory: true
+    description: Registration Key identical to the one previously configured on the device (`configure manager`).
+    example: key1
+    write_only: true
+    test_value: var.reg_key
+  - model_name: type
+    type: String
+    description: Type of the device; this value is always 'Device'.
+    exclude_example: true
+    requires_replace: true
+    default_value: Device
+    example: Device
+  - tf_name: access_policy_id
+    type: String
+    mandatory: true
+    model_name: id
+    data_path:
+      - accessPolicy
+    write_only: true # actually is read-write, but (for reading) generated code hurts custom code
+    description: The currently assigned access control policy.
+    example: fmc_access_control_policy.test.id
+    minimum_test_value: fmc_access_control_policy.minimum.id
+    test_value: fmc_access_control_policy.test.id
+  - tf_name: nat_policy_id
+    type: String
+    model_name: dummy_nat_policy_id
```

Comment:
To ease readability, let's make sure `model_name` is always the first element of the list entry.

Reply:
done, thanks